CONTRIBUTING.md

```diff
@@ -58,19 +58,130 @@ can be accepted.
 
 - Each new feature or change PR should provide meaningful CI tests.
 - All the tests and security scanning checks in CI should pass and achieve the expected coverage.
+- To avoid unnecessary failures in GitHub Actions, make sure `make check` and `make test` work locally.
+See below for instructions to set up the development environment.
 
 ### Merging PRs
 
-- Before a PR is merged, all commits in the PR should be rebased into meaningful commits.
+- Before a PR is merged, all commits in the PR should meaningful commits.
   - Its commit message should be the same as the PR title if there only one commit.
-- PRs should be merged using the fast-forward only strategy.
-  - This ensures a linear commit history for the project.
-  - If the fast-forward merge does not work due to new commits on `main` (in this case the two branches diverge and the PR branch cannot be fast-forwarded), the PR branch should be rebased onto `main` first.
+- PRs should be merged using the `Squash and merge` strategy. In most cases a single commit with
+a detailed commit message body is preferred. Make sure to keep the `Signed-off-by` line in the body.
 
 ## Branching model
 
 * The `main` branch is only used for releases and the `staging` branch is used for development. We only merge to `main` when we want to create a new release for Macaron.
 
+## Setting up the development environment
```

Comment:
Could you rebase on staging and update the reference to this section in `docs/README.md` - https://github.com/oracle-samples/macaron/blob/e6656c486f956e843e250d8dec3ace0bf36a2ecb/docs/README.md?plain=1#L7

Reply:
See commit [6de60e8](https://github.com/oracle-samples/macaron/pull/279/commits/6de60e887c592f5ef3b4bb984985148507222b1f)